core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -448,14 +449,16 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
   /**
    * Truncate the checkpointed offset for the given partition if its checkpointed offset is larger than the given offset
    */
-  def maybeTruncateCheckpoint(dataDir: File, topicPartition: TopicPartition, offset: Long): Unit = {
+  def maybeTruncateCheckpoint(dataDir: File, topicPartition: TopicPartition, offset: JLong): Unit = {
     inLock(lock) {
       if (logs.get(topicPartition).config.compact) {
         val checkpoint = checkpoints(dataDir)
         if (checkpoint != null) {
```

Comment:
this is unrelated to this PR, but the scala map will throw `NoSuchElementException` when there is no mapping value from the given key. Hence, the null check is redundant.

Reply:
There's a couple of places with similar code. I'd prefer changing these in a separate PR.